src/api/procedures/affirmConfidentialTransactions.ts

```diff
@@ -0,0 +1,138 @@
+import BigNumber from 'bignumber.js';
+
+import { PolymeshError, Procedure } from '~/internal';
+import {
+  ConfidentialAffirmParty,
+  ConfidentialLegProof,
+  ConfidentialTransaction,
+  ConfidentialTransactionStatus,
+  ErrorCode,
+  TxTags,
+} from '~/types';
+import { ExtrinsicParams, ProcedureAuthorization, TransactionSpec } from '~/types/internal';
+import {
+  confidentialAffirmsToRaw,
+  confidentialAffirmTransactionToMeshTransaction,
+} from '~/utils/conversion';
+
+interface SenderAffirm {
+  party: ConfidentialAffirmParty.Sender;
+  proofs: ConfidentialLegProof[];
+}
+
+interface ObserverAffirm {
+  party: ConfidentialAffirmParty.Mediator | ConfidentialAffirmParty.Receiver;
+}
+
+export type AffirmConfidentialTransactionParams = { legId: BigNumber } & (
+  | SenderAffirm
+  | ObserverAffirm
+);
```

Comment:
these all needs to be exported and also should be moved to types file to properly work with docs i guess